src/main/scala/play-json.scala

```diff
@@ -444,6 +444,83 @@ This can be caused by https://issues.scala-lang.org/browse/SI-7046 which can onl
     t
   }
 
+  def formatTagged[T: c.WeakTypeTag](enclosed: Tree)(tags: Tree): Tree = {
+    assertClass[T]()
+    val T = c.weakTypeOf[T]
+
+    val t =
+      q"""
+      {
+        import $pjson._
+        import $pkg._
+        new Format[$T] {
+          private val _delegateFormat: Format[$T] = $enclosed
+          private val _tags: Tags = implicitly[Tags]
+
+          def reads(json: JsValue): JsResult[$T] = {
+            (json \ _tags.field).asOpt[String] match {
+              case Some(tag) if _tags.isTagForClass(tag, classOf[$T]) => _delegateFormat.reads(json)
+              case Some(unexpectedTag) => JsError(s"Cannot deserialize type [$${classOf[$T]}] from json with type-tag [$${unexpectedTag}]")
+              case None => JsError(s"Cannot deserialize type [$${classOf[$T]}] from json without a type-tag field [_tags.field]")
+            }
+          }
+
+          def writes(value: $T): JsValue = _delegateFormat.writes(value) match {
+            case obj: JsObject => obj + (_tags.field, JsString(_tags.tagFor(classOf[$T])))
```

Comment:
can we add a check here, that this field does not already exist in `obj`? Otherwise it may lead to suprising bugs. A 4th case in the pattern match may be appropriate. Same check needed for formatSealedTagged